src/chainrules.jl

```diff
@@ -41,3 +41,15 @@ function ChainRulesCore.rrule(
     end
     return C, mul_pullback
 end
+
+function ChainRulesCore.rrule(
+    ::typeof(dot), A::AbstractSparseMatrix{T},B::AbstractSparseMatrix{T}
+) where T
+    r = dot(A, B)
+    function dot_pullback(r̄)
+        _, i_A, i_B = (~idot)(AD.GVar(r, r̄), AD.GVar(A), AD.GVar(B))
```

Comment:
Curious to ask, do we need `r̄ = unthunk(r̄)` here?

Reply:
In fact, only ` function ChainRulesCore.rrule(::typeof(*), A::AbstractSparseMatrix{T}, B::DenseInputVecOrMat
) where T ` need `unthunk`, all the other methods passed tests on my local machine without `unthunk`. Adding `unthunk` for all the `rrule` is a safer way, I think.